.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,20 @@
+name: Style Check
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  clang-format:
+    name: Clang-Format
```

Comment:
Could you try to add a [`permission:` keyword](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token) with as little permissions as possible to run this action?

Currently, this is not necessary. If we switch to restrictive default permissions via #1737 though, we will need this. Maybe there are other users of this action where we could borrow the right set of permissions?

Reply:
I have restricted the action now to
```
    permissions:
      contents: read
```

Reply:
Sounds good, thanks! :)